routers/auth-router.js

```diff
@@ -0,0 +1,86 @@
+const { checkDuplicateUsername, checkRolesExist } = require("../middlewares/verify-signup");
+const config = require("../config/auth-config");
+const { User } = require("../models/user-model")
+const { Role } = require("../models/role-model")
+var jwt = require("jsonwebtoken");
+var bcrypt = require("bcryptjs");
+const express = require("express")
+const router = express()
+
+router.route("/signup")
+.post(async (req, res) => {
+  try {
+    const user = req.body
+    const newUser = new User({ name: user.name, username: user.username, password: bcrypt.hashSync(req.body.password, 8)});
+    const doesUserExist = await User.findOne({username: newUser.username})
+    if(doesUserExist) {
+      console.log("username exists")
+      return res.json({ success: false, message: "Username already in use, try another"})
+    }
+    const savedUser = await newUser.save();
+    if(newUser.role) {
+        const role = await Role.find({name: {$in: newUser.role}}, (err, roles) => {
+            if(err) {
+                return res.status(500).json({success: false, message: "error occurred while finding roles", errMessage: err.message})
+            }
+            newUser.role = role.map(currentRole => currentRole._id)
+            newUser.save();
+            res.status(200).json({success: true, message: "user registered successfully", user: savedUser})
+        })
+    } else {
+        const role = await Role.findOne({name: "buyer"}, (err, role) => {
+            if(err) {
+                return res.status(500).json({success: false, message: "error occurred while adding role", errMessage: err.message})
+            }
+            newUser.role = [role._id];
+            console.log(user.role)
+            newUser.save();
+            res.status(200).json({success: true, message: "user registered successfully", user: savedUser})
+        } )
+    }
+    console.log(savedUser);
+  } catch (err) {
+      console.log("error occurred while signing up")
```

Comment:
The error can be console logged here as it will be helpful when checking the log files

Reply:
okay!